include/xtensor/xutils.hpp

```diff
@@ -891,6 +892,85 @@ namespace xt
      */
     template <class T>
     using squared_norm_type_t = typename squared_norm_type<T>::type;
+
+    namespace alloc_tracking
+    {
+        static bool enabled = false;
+
+        static void enable()
+        {
+            enabled = true;
+        }
+
+        static void disable()
+        {
+            enabled = false;
+        }
+    }
```

Comment:
I think tracking allocator should always been enabled, we don't need to toggle it dynamically. Besides, here your static variable "enabled" would be local to the translation unit including this header (the same for the functions), I wonder what the behavior would be when included in different .cpp.

Reply:
Actually it may be useful to track allocations in a given scope. In that case, the `enabled` variable should be a static member of the `tracking_allocator` class instead, and the enabling/disabling functinos should be available in the `xt` namespace (their implementation binding on their counterpart method in `tracking_allocator`)

Reply:
Ok, I'll try to refactor that.

Reply:
ok, I tried, but I don't think it's possible.
Having the `bool enabled` as part of the allocator would have the user need to know for which combination of template types he wants to activate tracking etc. 

So I believe going with the global bool in a namespace is the way to go.